### PR TITLE
Example for conversion ERC20 to XC20

### PR DESCRIPTION
**Pull request by Maar-io**
An example for conversion ERC20 to XC20.
Added Smart contract `xcBurrito` and simple instructions how to do the conversion

**Check list**
- [ ] updated documentation

### CODE REVIEW

**Review by niklabh — APPROVED** [review]

**Review by andabak — COMMENTED** [review]
So suggestion would be to move this content to the docs.astar.network instead of using it here - since it is not a working code but just suggestion how the code could work?

File: precompiles/assets-erc20/example/README.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# How to Convert Existing ERC20 Token Into XC20 Asset
+This an example how you can enhance already deployed ERC20 token to cross-chain-ready XC20 asset.
+
```
Comment (Dinonard) [inline precompiles/assets-erc20/example/README.md]: Perhaps we should first describe the scenario at which this example is aimed at.

File: precompiles/assets-erc20/example/README.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+To deploy XcBurrito contract you need 2 input parameters
+- Burrito ERC20 token address (H160)
+- XC20 asset address (H160)
+    - Follow instructions on how to [Generate XC20 Address](https://docs.astar.network/docs/xcm/integration/tools). If your asset_id is 17 (=0x11), your EVM address will be `0xffffffff00000000000000000000000000000011`
+
+### 3. Transfer xc asset ownership to the XcBurrito smart contract
+To enable xcBurrito contract to be able to mint/burn you need to call `setTeam()` extrinsic on pallet-assets. Since the EVM contract has 24 bytes (H160), we need to transform EVM address into SS58 format to be able to call this extrinsic.
+<img width="1205" alt="Screenshot 2022-08-04 at 09 32 11" src="https://user-images.githubusercontent.com/34627453/182795050-f10dad6e-dbd9-4ac9-b526-d51a66503cb9.png">
+After you set the team you should renounce the ownership over the xc20 asset by calling `transferOwnership()` extrinsic on pallet-assets.
+
```
Comment (Dinonard) [inline precompiles/assets-erc20/example/README.md]: Maybe this should be worded a bit differently - it's not really critical for the technical functionality but it is for the decentralization.

File: precompiles/assets-erc20/example/README.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+The result for `withdrawTo(user, amount)` will be:
+- XcBurrito contract burns `amount` of XcBurrito asset
+- User's balance for XcBurrito asset is decreased for `amount`
+- XcBurrito contract sends `amount` of ERC20 Burrito token to user
+- User's balance for ERC20 Burrito is increased for `amount`
+
+
+
```
Comment (Dinonard) [inline precompiles/assets-erc20/example/README.md]: For the finish, we should state that now wrapped token can be sent to other parachains.

**Review by Dinonard — COMMENTED** [review]
Forgot to add general comments:
* we should move readme on to astar-docs
* perhaps it's good to create an `examples` folder in repo root and put all related code there (I guess we'll have more of those in the future)

File: precompiles/assets-erc20/example/README.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+This is useful for several reasons, to mention a few:
+* To extend the functionalities of an existing token in conjunction with other ERC20 modules.
+* Allow a native cryptocurrency to behave like an ERC20, e.g. Wrapped ether (WETH).
+* Allow the use of currencies outside its native blockchain, e.g. Wrapped bitcoin (WBTC).
+
+In this example we'll do a mix of 1st and 3rd use case. We are wrapping existing ERC20 token to become XC20.
+
+## Wrapping ERC20 example
+Underlaying token will be existing ERC20, let's name it BURRITO. We want to wrap this BURRITO token and make it cross-chain-ready. Using standard [ERC20Wrapper](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/extensions/ERC20Wrapper.sol) token spec from OpenZeppelin will not be enough. Therefore we will override some of the  ERC20Wrapper functions to use XC20+ functions.
```
Comment (andabak) [inline precompiles/assets-erc20/example/README.md]: @Dinonard - I am working on moving this to docs.astar (https://github.com/AstarNetwork/astar-docs/tree/feature/erc20-to-xc20-new-doc) but when I read this, this is actually the first place where XC20+ (note the '+') is mentioned in the docs without any previous reference. Do you think we should write some sub-paragraph on XC20+ (vs XC20) or perhaps extend our XC20 page in docs to elaborate what XC20+ is (https://docs.astar.network/docs/xcm/building-with-xcm/create-xc20-assets)

Comment (Dinonard) [reply precompiles/assets-erc20/example/README.md]: Sorry, just read this - I've changed the text a bit and prepared the branch.

Let me check about the question.

**Pull request merged**